client/services/FeedbackGeneratorService.js

```diff
@@ -208,11 +207,39 @@ tie.factory('FeedbackGeneratorService', [
           }
         }
       );
-      feedback.appendCodeParagraph(fixedErrorString);
+
+      var feedbackString = _generateRuntimeFeedback(fixedErrorString);
+      if (feedbackString === "") {
+        feedback.appendTextParagraph(
+            "Looks like your code had a runtime error" + inputClause +
+            ". Here's the trace:");
+        feedback.appendCodeParagraph(fixedErrorString);
+      } else {
+        feedback.appendTextParagraph(feedbackString);
+      }
       return feedback;
     };
 
     /**
+     * Based on passed in error string, will generate the appropriate,
+     * informative feedback to be appended to the overall submission feedback.
+     *
+     * @param {string} errorString
+     * @returns {string} Text to be appended to feedback.
+     */
+    var _generateRuntimeFeedback = function(errorString) {
```

Comment:
Maybe call this _getHumanReadableRuntimeFeedback()? Since the emphasis here is on getting a human-understandable message